components/admin_console/admin_definition.jsx

```diff
@@ -3395,11 +3395,36 @@ export default {
                         help_text: t('admin.guest_access.mfaDescription'),
                         help_text_default: 'When true, [multi-factor authentication](!https://docs.mattermost.com/deployment/auth.html) for guests is required for login. New guest users will be required to configure MFA on signup. Logged in guest users without MFA configured are redirected to the MFA setup page until configuration is complete.\n \nIf your system has guest users with login methods other than AD/LDAP and email, MFA must be enforced with the authentication provider outside of Mattermost.',
                         help_text_markdown: true,
-                        isDisabled: it.either(
+                        isHidden: it.either(
                             it.configIsFalse('ServiceSettings', 'EnableMultifactorAuthentication'),
                             it.configIsFalse('ServiceSettings', 'EnforceMultifactorAuthentication'),
                         ),
                     },
+                    {
+                        type: Constants.SettingsTypes.TYPE_BOOL,
+                        key: 'GuestAccountsSettings.EnforceMultifactorAuthentication',
+                        label: t('admin.guest_access.mfaTitle'),
+                        label_default: 'Enforce Multi-factor Authentication: ',
+                        help_text: t('admin.guest_access.mfaDescriptionMFANotEnabled'),
+                        help_text_default: '[multi-factor authentication](./mfa) is currently not enabled.',
+                        help_text_markdown: true,
+                        isDisabled: () => true,
+                        isHidden: it.configIsTrue('ServiceSettings', 'EnableMultifactorAuthentication'),
+                    },
+                    {
+                        type: Constants.SettingsTypes.TYPE_BOOL,
+                        key: 'GuestAccountsSettings.EnforceMultifactorAuthentication',
+                        label: t('admin.guest_access.mfaTitle'),
+                        label_default: 'Enforce Multi-factor Authentication: ',
+                        help_text: t('admin.guest_access.mfaDescriptionMFANotEnforced'),
+                        help_text_default: '[multi-factor authentication](./mfa) is currently not enforced.',
```

Comment:
@wiersgallak I added the case where MFA is not enforced, so we can't enable MFA Enforcement for guest accounts. Is that correct? It wasn't defined in the UX.

Reply:
@jespino yes that would be my expectations. Thanks for catching.